spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskManifest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.core;
+
+import java.util.List;
+
+import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
+import org.springframework.core.style.ToStringCreator;
+
+/**
+ * Description of an execution of a task including resource to be executed and how it was configured via Spring Cloud
+ * Data Flow
+ *
+ * @author Mark Pollack
+ * @author Michael Minella
+ * @since 2.3
+ */
+public class TaskManifest {
```

Comment:
Along with the task deployment requests, it may be nice to add the `taskDeploymentId` into the task manifest as it is obtained only after launching the task and would be handy to persist as part of the `TaskManifest`.

Reply:
What do you mean by `taskDeploymentId`?  The task execution id?  The external execution id?  something else?

Reply:
I meant the `taskDeploymentId` for the [TaskDeployment](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDeployment.java#L39) which is the external execution ID as well.

Reply:
Can we scope that as an enhancement?  You can already associate one with the other since each manifest is associated with a task execution (which has that) anyways...

Reply:
Sure! Created https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/1304 to track this.

Reply:
Created the issue in the wrong repo. Corrected: https://github.com/spring-cloud/spring-cloud-dataflow/issues/3522